src/v/kafka/server/replicated_partition.cc

```diff
@@ -314,10 +314,17 @@ ss::future<error_code> replicated_partition::prefix_truncate(
       || kafka_truncation_offset > high_watermark()) {
         co_return error_code::offset_out_of_range;
     }
-    const auto rp_truncate_offset = _translator->to_log_offset(
-      kafka_truncation_offset);
+    model::offset rp_truncate_offset{};
+    auto local_kafka_start_offset = _translator->from_log_offset(
+      _partition->raft_start_offset());
+    if (kafka_truncation_offset > local_kafka_start_offset) {
```

Comment:
does this logic also apply when portions of the local log are not all available locally and the rest is in TS?